### ISSUE TITLE
changing vm-driver from hyperkit to another should delete left over

### ISSUE THREAD

**Issue by medyagh** [kubernetes/minikube#4892]
After user decides to replace hyperkit with another driver, we should delete the left over hyperkit.

to reproduce:

- minikube start --vm-driver=hyperkit

- minikube start --vm-driver=virtualbox

- pgrep hyperkit

the last one should return no more than 1 pid ! 

@josedonizetti do you have ideas how to tackle this ?

**Comment by RA489** [comment]
@medyagh Is it happening only for hyperkit with another driver or with another vm-drivers too?

**Comment by josedonizetti** [comment]
@medyagh Wonder if you got a different output here? When I run two `start` commands with different drives sequentially I get:

```
minikube start --vm-driver=virtualbox
😄  minikube v1.2.0 on darwin (amd64)


⚠️  Ignoring --vm-driver=virtualbox, as the existing "minikube" VM was created using the hyperkit driver.
⚠️  To switch drivers, you may create a new VM using `minikube start -p <name> --vm-driver=virtualbox`
⚠️  Alternatively, you may delete the existing VM using `minikube delete -p minikube`


🏃  Re-using the currently running hyperkit VM for "minikube" ...
⌛  Waiting for SSH access ...
🐳  Configuring environment for Kubernetes v1.15.0 on Docker 18.09.6
🔄  Relaunching Kubernetes v1.15.0 using kubeadm ...
⌛  Verifying: apiserver proxy etcd scheduler controller dns
🏄  Done! kubectl is now configured to use minikube
```

same thing for kvm2/linux

**Comment by josedonizetti** [comment]
What I think is the problem here is that if we do:

```
./out/minikube start --vm-driver=kvm2
./out/minikube profile list
|---------|-----------|---------------|-----------|--------------------|
| Profile | VM Driver |    NodeIP     | Node Port | Kubernetes Version |
|---------|-----------|---------------|-----------|--------------------|
| test    | kvm2      | 192.168.39.56 |      8443 | v1.15.0            |
|---------|-----------|---------------|-----------|--------------------|
```

then do a clean start using the virtualbox driver:

```
 ./out/minikube start --vm-driver=virtualbox
./out/minikube profile list
|---------|------------|---------------|-----------|--------------------|
| Profile | VM Driver  |    NodeIP     | Node Port | Kubernetes Version |
|---------|------------|---------------|-----------|--------------------|
| test    | virtualbox | 192.168.39.56 |      8443 | v1.15.0            |
|---------|------------|---------------|-----------|--------------------|
```

We get the vm-driver as `virtualbox` although this vm is still on kvm2:

```
virsh list
 Id    Name                           State
----------------------------------------------------
 10    test                           running
````

What we need to do here is not set the vm-driver to `virtualbox` on the second start.

@medyagh ???

**Comment by tstromberg** [comment]
I believe this was addressed in minikube v1.4, which refuses to hot-swap between drivers.
